core/templates/dev/head/editor/StateResponses.js

```diff
@@ -318,8 +328,9 @@ oppia.controller('StateResponses', [
       var hasFeedback = outcome.feedback.some(function(feedbackItem) {
         return Boolean(feedbackItem);
       });
-
-      return isSelfLoop(outcome) && !hasFeedback;
+      if (!$scope.suppressDefaultAnswerGroupWarnings()) {
```

Comment:
I don't think this is the place to add this check, since it doesn't hew with the name of the method. Instead, if something in the HTML makes use of isSelfLoopWithNoFeedback(), and we want it to also take into account suppressDefaultAnswerGroupWarnings(), then that place should refer to the latter method explicitly.